src/Commands/BuildToolsBase.php

```diff
@@ -343,41 +344,64 @@ protected function expandSourceAliases($source)
     }
 
     /**
-     * Detect the upstream to use based on the contents of the source repository.
-     * Upstream is irrelevant, save for the fact that this is the only way to
-     * set the framework on Pantheon at the moment.
+     * Detect the application and version at a given directory.
+     *
+     * @todo Replace determining the framework from the upstream with this.
      */
-    protected function autodetectUpstream($siteDir)
+    protected function autodetectApplication($siteDir)
     {
-        $upstream = $this->autodetectUpstreamAtDir("$siteDir/web");
-        if ($upstream) {
-            return $upstream;
+        $info = $this->autodetectApplicationAtDir("$siteDir/web");
+        if ($info) {
+            return $info;
         }
-        $upstream = $this->autodetectUpstreamAtDir($siteDir);
-        if ($upstream) {
-            return $upstream;
+        $info = $this->autodetectApplicationAtDir($siteDir);
+        if ($info) {
+            return $info;
         }
-        // Can't tell? Assume Drupal 8.
-        return 'Empty Upstream';
+
+        return false;
     }
 
-    protected function autodetectUpstreamAtDir($siteDir)
+    protected function autodetectApplicationAtDir($siteDir)
     {
-        $upstream_map = [
-          'core/misc/drupal.js' => 'empty', // Drupal 8
-          'misc/drupal.js' => 'empty-7', // Drupal 7
-          'wp-config.php' => 'empty-wordpress', // WordPress
-        ];
+        $version_info = new VersionTool();
+        $info = $version_info->info($siteDir);
 
-        foreach ($upstream_map as $file => $upstream) {
-            if (file_exists("$siteDir/$file")) {
-                return $upstream;
-            }
+        if ($info) {
+            $application = $info->application();
+            $version = $info->version();
+            $major_version = explode(".", $version);
+            $framework = $application == "WordPress" ? "empty-wordpress" : (
```

Comment:
This could be a little more verbose for readability. Maybe break it out into its own function where you can use the pattern `if (cond) { return }` to avoid nesting and ternary operators.